src/Apps/Artist/Routes/Overview/Components/RecommendedArtist.tsx

```diff
@@ -0,0 +1,188 @@
+import { EntityHeader, Sans, Spacer } from "@artsy/palette"
+import { RecommendedArtist_artist } from "__generated__/RecommendedArtist_artist.graphql"
+import { track } from "Artsy/Analytics"
+import * as Schema from "Artsy/Analytics/Schema"
+import { SystemContext } from "Artsy/SystemContext"
+import { FillwidthItem } from "Components/Artwork/FillwidthItem"
+import { FollowArtistButtonFragmentContainer as FollowArtistButton } from "Components/FollowButton/FollowArtistButton"
+import { Carousel } from "Components/v2"
+import { stringify } from "qs"
+import React, { FC, useContext } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { data as sd } from "sharify"
+import { get } from "Utils/get"
+
+interface RecommendedArtistProps {
+  artist: RecommendedArtist_artist
+}
+const HEIGHT = 150
+
+// TODO: Remaining tasks
+/*
+ * find out tracking constants to use
+ * Move handleOpenAuth/openMobileAuth/openDesktopAuth to shared location
+ * Buy now/make offer labels
+ * Possibly extract <followArtistButton> to a shared location (it was a lot of copypasta that just worked)
+ * Tests
+ */
+
+const handleOpenAuth = (mediator, artist) => {
+  if (sd.IS_MOBILE) {
+    openMobileAuth(artist)
+  } else if (mediator) {
+    openDesktopAuth(mediator, artist)
+  } else {
+    window.location.href = "/login"
+  }
+}
+
+const openMobileAuth = artist => {
+  const params = stringify({
+    action: "follow",
+    contextModule: Schema.ContextModule.RecommendedArtists,
+    intent: "follow artist",
+    kind: "artist",
+    objectId: artist.id,
+    signUpIntent: "follow artist",
+    trigger: "click",
+    entityName: artist.name,
+  })
+  const href = `/sign_up?redirect-to=${window.location}&${params}`
+
+  window.location.href = href
+}
+
+const openDesktopAuth = (mediator, artist) => {
+  mediator.trigger("open:auth", {
+    mode: "signup",
+    copy: `Sign up to follow ${artist.name}`,
+    signupIntent: "follow artist",
+    afterSignUpAction: {
+      kind: "artist",
+      action: "follow",
+      objectId: artist.id,
+    },
+  })
+}
+
+@track({
+  context_module: Schema.ContextModule.RecommendedArtists,
+})
+class RecommendedArtistWithTracking extends React.Component<
```

Comment:
Any thoughts on this "pattern"? I pulled the tracking stuff out of the `RecommendedArtist` component below (https://github.com/artsy/reaction/pull/2255/files#diff-b7e8e47a9fac963f0852e544e7e80e51R92), because the `@track` calls were the only thing that required a class. 

This allowed me to keep the `RecommendedArtist` component functional, and to use `useContext` instead of `ContextConsumer`/render props. It's kind of a "container & presentational" breakdown, I guess.

(And maybe there's a way to apply the `@track` calls to a function, instead of a class with methods, but it didn't look like it to me. @damassi what do you think?)

Reply:
Looks cool to me! I don't have a strong opinion, I possibly would have just gone with a class overall. Personally I pretty much always go w/ a React class over FC/SFC component, since a lot of times I wind up wanting to add a bit of local state. That's been much more useful (to me), vs having more functional components. Tests wind up being more integration-y (or snapshots) anyway, and I haven't needed to do any fancy functional....stuff.

So while I probably would have just had everything in one React class, splitting a piece of it out as you've done is 👍 from me nonetheless!

Reply:
Now that we've introduced hooks to Reaction, my plan is to use them whenever possible moving forward. I'm avoiding classes as much as I can.